### PR TITLE
Feature/3439 app bar playground

### LINKED ISSUE #508: AppBar not listening for scroll events when rendered as a ng-template
#### Describe the bug
AppBar not listening for scroll events when rendered as a ng-template.
An attempt to listen for a scroll container is happening before the content is rendered on the screen.

#### What is the expected behavior?
Scroll element should be found by the Appbar component so snap variant works as expected.

#### What are the steps to reproduce?
1.  Create a `ng-template` with an app bar inside of it. (#appbar)
2.  Pass it into a template `<template [ngTemplateOutlet]="appbar">`
3. Observe `snap` variant does not work.

#### Screenshots or links to minimum reproduction example
scrolllll

#### Environment

Chrome Windows	
	
#### Anything else to add?
No

### PR DESCRIPTION
**Pull request by emclaug2**
#### Changes proposed in this Pull Request:
- AppBar Playground

#### Fixes
fixes #508

### CODE REVIEW

File: src/docs/pages/component-docs/components/app-bar/examples/playground.component.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            <button blui-icon mat-icon-button style="margin: 0 24px 0 -8px">
+                <mat-icon>menu</mat-icon>
+            </button>
+            <blui-three-liner title="Title" subtitle="Subtitle" info="Info" [style.top.px]="isCollapsed ? 0 : 80">
+            </blui-three-liner>
+            <div blui-actions style="display: flex; margin: 0 -8px;">
+                <mat-icon style="margin: 0 8px">search</mat-icon>
+                <mat-icon style="margin: 0 8px">download</mat-icon>
+                <mat-icon style="margin: 0 8px">more_vert</mat-icon>
+            </div>
```
[the remark this answers is not in the document]
Comment (emclaug2) [reply src/docs/pages/component-docs/components/app-bar/examples/playground.component.ts:50]: Very cool find, I've updated the example to use gap.

File: src/docs/pages/component-docs/components/app-bar/app-bar-doc.component.ts [hunk 2]
```diff
@@ -101,4 +115,47 @@ export class AppBarDocComponent {
...
+        collapsedHeight: {
+            value: 64,
+            type: 'number',
+            hint: 'Height of the AppBar when collapsed',
+            range: { min: 56, max: 100, step: 1, tickInterval: 1 },
+        },
+        scrollThreshold: {
+            value: 100,
+            type: 'number',
+            hint: 'Height of the AppBar when collapsed',
```
[the remark this answers is not in the document]
Comment (emclaug2) [reply src/docs/pages/component-docs/components/app-bar/app-bar-doc.component.ts:136]: Ooof, good catch.

**Comment by emclaug2** [comment]
> I was also a little surprised that we do not have a `'default'` color for blui-app-bar

The default color is supposed to be `primary`, I didn't create the knobs correctly.   Should be good now!

**Comment by emclaug2** [comment]
> The button ripple is not centered

I'm having a hard time reproducing this one.  The icon is off-centered, for me, based on how far out I'm zoomed.

**Comment by huayunh** [comment]
... and you need to update the playground code

image

**Review by huayunh — APPROVED** [review]
Ran out of complaints

**Comment by emclaug2** [comment]
> Ran out of complaints

🚀 Going to merge this one in.  Thanks @huayunh , good observations

**Pull request merged**